apps/angular/src/app/features/anime/pages/dashboard/dashboard-page.component.ts

```diff
@@ -43,12 +90,125 @@ export class DashboardPageComponent {
 		this.animes$ = this.createAnimeStream();
 	}
 
+	/** @inheritdoc */
+	public ngOnInit(): void {
+		this.searchValue.set(this.queryParams.search);
+	}
+
+	/**
+	 * Get anime list.
+	 * @param params Query params.
+	 */
+	private getAnimeList(params: AnimeParams): Observable<Pagination<Anime>> {
+		return this.animeService.getAnimeList(params);
+	}
+
 	private createAnimeStream(): Observable<Pagination<Anime>> {
-		this.isLoading$.next(true);
+		this.isLoading.set(true);
 
-		return this.animeService.getAnimeList().pipe(
-			takeUntilDestroyed(this.destroyRef),
-			finalize(() => this.isLoading$.next(false)),
+		return this.activeRoute.queryParams.pipe(
+			distinctUntilChanged(),
+			map(query => this.transformQueryParams(query)),
+			switchMap(params =>
+				this.getAnimeList(params).pipe(
+					finalize(() => {
+						this.isLoading.set(false);
+						this.selectedTypes.set(this.transformTypes(this.queryParams.typeIn));
+					}),
+					takeUntilDestroyed(this.destroyRef),
+				)),
 		);
 	}
+
+	/**
+	 * Transform type parameters.
+	 * @param typeIn Query parameters.
+	 */
+	private transformTypes(typeIn?: string): AnimeType[] {
+		if (!typeIn) {
+			return [];
+		}
+
+		return typeIn.split(',') as AnimeType[];
+	}
+
+	/**
+	 * Transform query parameters.
+	 * @param query Query parameters.
+	 */
+	private transformQueryParams(query: Params): QueryParams {
```

Comment:
For interaction with API we use the service, but for Query parameters the same operations are placed directly in the component, I see inconsistency here